test/generator/binary_protocol_test.exs

```diff
@@ -221,6 +222,53 @@ defmodule Thrift.Generator.BinaryProtocolTest do
     assert_serializes %Struct{val_list: [%Val{num: 91}]},             <<15, 0, 4, 12, 0, 0, 0, 1, 3, 0, 99, 91, 0, 0>>
   end
 
+  @thrift_file name: "unions.thrift", contents: """
+  struct StructValue {
+    1: string username;
+  }
+  union Union {
+    1: i64 int_field,
+    2: StructValue struct_field,
+    3: string string_field,
+    4: list<i16> list_field;
+  }
+
+  struct UStruct {
+    1: Union my_union,
+    2: list<Union> u_list,
+    3: set<Union> u_set,
+    4: map<Union, Union> u_map,
+  }
```

Comment:
It's helpful when scanning the code with a question in mind like, did we test all the combinations? Val, map, set, list, val, map, set, list...

I can see how it would come across as OCD, but consistency like that is kind to future maintainers of the code.